optimum/neuron/utils/training_utils.py

```diff
@@ -315,6 +315,28 @@ def skip_first_batches(dataloader, num_batches=0):
     return dataloader
 
 
+@requires_neuronx_distributed
+@requires_safetensors
+def torch_xla_safe_save_file(
+    tensors: Dict[str, torch.Tensor],
+    filename: Union[str, os.PathLike],
+    metadata: Optional[Dict[str, str]] = None,
+    master_only: bool = True,
+    global_master: bool = False,
+):
+    """
+    Torch XLA compatible implementation of `safetensors.torch.save_file`.
```

Comment:
Could you elaborate and explain those arguments a bit? I am not familiar with what `master_only` and `global_master` indicate.

Reply:
So those two parameters are related to distributed training.
Basically,  when `master_only` is `True`, only the master rank will be saving the file instead of all ranks, and `global_master` controls whether only the global master (accross multiple nodes) should be saving or the master of each node.

Reply:
More information [here](https://github.com/pytorch/xla/blob/30458aaae73a496f73c29bb468b04c283877217f/torch_xla/utils/serialization.py#L61-L69)